formapp/app.py

```diff
@@ -1718,6 +1764,26 @@ def endpoint_sendblock():
         return "Bad Arguments. Requires 'id' and 'data'"
 '''
 
+#Choose whether you want to use basic or expert mode
+@app.route('/setdefault', methods = ['POST', 'GET'])
+def endpoint_setdefault():
+    if not request.remote_addr in sD.studentDict:
+        return redirect('/login?forward=' + request.path)
+    if request.method == 'POST':
+        if request.form['mode'] == 'basic' or request.form['mode'] == 'advanced':
+            db = sqlite3.connect(os.path.dirname(os.path.abspath(__file__)) + '/data/database.db')
+            dbcmd = db.cursor()
+            sD.studentDict[request.remote_addr]['preferredMode'] = request.form['mode']
+            ##Test this
+            dbcmd.execute("UPDATE users SET preferredMode=:mode WHERE username=:uname", {"uname": sD.studentDict[request.remote_addr]['name'], "mode": request.form['mode']})
+            db.commit()
+            db.close()
+        else:
+            return 'Invalid mode.'
+        return redirect('/')
+    else:
+        return render_template('setdefault.html')
+
```

Comment:
To reduce the number of possible endpoints, can this be merged with another endpoint and just use a query param to handle this code?